src/utilities/modules/dispensary/refund.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import { createRecord } from '../../../database/utilities';
+import { UIDatabase } from '../../../database';
+
+/**
+ * Utility to refund a collection of TransactionBatch records. Creates a
+ * single CustomerCredit and for each TransactionBatch that should be
+ * refund, a RefundLine record, which enters the batch back into stock
+ * and adds credit for the patient.
+ *
+ *
+ * @param {User}               currentUser The currently logged in user.
+ * @param {Name}               patient     The patient to refund to
+ * @param {TransactionBatch[]} batches     Array of batches to be refund
+ */
+export const refund = (currentUser, patient, batches) => {
+  const forSamePatient = batches.every(({ transaction }) => {
+    const { otherPartyName } = transaction;
+    return patient.id === otherPartyName.id;
```

Comment:
`otherPartyName` couldn't be null, could it?

Reply:
oops thats broken completely, cheers - e375cbae25c5ae62762dd3a0d19651c41e921df8